### PR TITLE
Changes:

### PR DESCRIPTION
**Pull request by psyoptix**
Changes in this PR
---------------------
- Move use_account & use_domain bools from X509Parameters to IdentityConfiguration
- Generic ID provider must define method to extract account and/or domain from a cred.
  = add extract functions to example interface of id provider
  = ...which we instantiate for the X509 ID Provider using Common Names in cert chain.
- Identity gets its own section.
- Gateway becomes subsection of intro.
- Operational context becomes subsection of intro.

Still ToDo after this PR
------------------------
- Gateway should know its own domain and do sanity checks when adding keypackages
- Gateway should sanity check when fetching remote key packages (from another domain).
- In Op. Context section give example flows
  = Client registration, KP registration, fetch KP, issue commit & welcome.

### CODE REVIEW

File: draft-aegis-mimi-arch-latest.md [hunk 1]
```diff
@@ -145,25 +166,27 @@ interface {
...
 
-Each identity provider has a unique value that identifies its behavior along
-with opaque parameters which help synchronize various options across
-applications. 
+If the provider is configured to support accounts the interface could, for
+example, include the following additional function.
 
 ~~~
-uint16 IdentityProviderType;
+    fn client_account(Identity identity) -> Vec<u8>
```
Comment (tomleavy) [inline draft-aegis-mimi-arch-latest.md]: If accounts / domain are constrained to a subset of UTF8 character encoding I think this should be `String`
```suggestion
    fn client_account(Identity identity) -> String
```

File: draft-aegis-mimi-arch-latest.md [hunk 2]
```diff
@@ -145,25 +166,27 @@ interface {
...
 
-struct {
-    IdentityProviderType type;
-    opaque parameters<V>;
-} IdentityConfiguration;
+Similarly, when configured to support domains the interface could be extended
+with the following function.
+
+~~~
+    fn client_domain(Identity identity) -> Vec<u8>
```
Comment (tomleavy) [inline draft-aegis-mimi-arch-latest.md]:
```suggestion
    fn client_domain(Identity identity) -> String
```

File: draft-aegis-mimi-arch-latest.md [hunk 3]
```diff
@@ -196,25 +219,24 @@ struct {
 } X509DeviceHandleRange
 
 struct {
-    bool use_accounts;
-    bool use_domains:
     X509DeviceHandleRange device_handle_range;
-    select (use_accounts) {
+    select (supports_accounts) {
```
Comment (tomleavy) [inline draft-aegis-mimi-arch-latest.md]:
```suggestion
    select (IdentityConfiguration.supports_accounts) {
```

File: draft-aegis-mimi-arch-latest.md [hunk 4]
```diff
@@ -196,25 +219,24 @@ struct {
...
         case true:
             uint8 account_handle_offset;
         case false:
             struct {}
     }
-    select (use_domains) {
+    select (supports_domains) {
```
Comment (tomleavy) [inline draft-aegis-mimi-arch-latest.md]:
```suggestion
    select (IdentityConfiguration.supports_domains) {
```

File: draft-aegis-mimi-arch-latest.md [hunk 5]
```diff
@@ -227,22 +249,22 @@ handel offset MUST strictly succeed the device handle range in the
 certificate chain counting from the leaf certificate up. 
 
 ~~~
-If (params.use_accounts == true) {
+If (conf.supports_accounts == true) {
```
Comment (tomleavy) [inline draft-aegis-mimi-arch-latest.md]:
```suggestion
If (IdentityConfiguration.supports_accounts == true) {
```

File: draft-aegis-mimi-arch-latest.md [hunk 6]
```diff
@@ -227,22 +249,22 @@ handel offset MUST strictly succeed the device handle range in the
...
+Account handles are calculated like client handles but based on the certificate
 at the Account Handle offset (instead of Client Handle range).
 
 If the provider is configured to use domains then the domain name offset MUST
 strictly succeeds the device handle range in the certificate chain counting
 from the leaf certificate up.
 
 ~~~
-If (params.user_domains == true) {
+If (conf.supports_domains == true) {
```
Comment (tomleavy) [inline draft-aegis-mimi-arch-latest.md]:
```suggestion
If (IdentityConfiguration.supports_domains == true) {
```

File: draft-aegis-mimi-arch-latest.md [hunk 7]
```diff
@@ -251,7 +273,7 @@ Finally, if both domains and accounts are used then the domain name offset MUST
 also strictly succeed the account handle offset.
 
 ~~~
-If (params.use_domains == true) && (params.use_accounts == true) {
+If (conf.supports_domains == true) && (conf.supports_accounts == true) {
```
Comment (tomleavy) [inline draft-aegis-mimi-arch-latest.md]:
```suggestion
If (IdentityConfiguration.supports_domains == true) && (IdentityConfiguration.supports_accounts == true) {
```

File: draft-aegis-mimi-arch-latest.md [hunk 8]
```diff
@@ -269,7 +291,7 @@ seperated by a "/". When accounts are not used the client handle is simply its
 device handle.
 
 ~~~
-If (params.use_accounts == true) {
+If (conf.supports_accounts == true) {
```
Comment (tomleavy) [inline draft-aegis-mimi-arch-latest.md]:
```suggestion
If (IdentityConfiguration.supports_accounts == true) {
```

**Review by tomleavy — COMMENTED** [review]
Minor cleanup

**Review by mulmarta — APPROVED** [review]

**Pull request merged**